src/portal/mod.rs

```diff
@@ -283,24 +284,45 @@ impl Keyring {
         }
         Ok(())
     }
-}
 
-/// Return key, derive and store it first if not initialized
-async fn derive_key<'a>(
-    key: &'a mut OnceCell<crate::Key>,
-    keyring: &api::Keyring,
-    secret: &Secret,
-) -> &'a crate::Key {
-    key.get_or_init(|| keyring.derive_key(secret))
+    /// Return key, derive and store it first if not initialized
+    async fn derive_key<'a>(&'a self, key: &'a mut OnceCell<crate::Key>) -> &'a crate::Key {
+        let keyring = Arc::clone(&self.keyring);
+        let secret = Arc::clone(&self.secret);
+
+        #[cfg(feature = "async-std")]
+        let newkey = async_global_executor::spawn_blocking(move || {
+            async_std::task::block_on(async { keyring.read().await.derive_key(&secret) })
+        })
+        .await;
+
+        #[cfg(feature = "tokio")]
+        let newkey =
+            tokio::task::spawn_blocking(move || keyring.blocking_read().derive_key(&secret))
+                .await
+                .unwrap();
```

Comment:
is it safe to unwrap here?

Reply:
I mean it fails if the thread panics which would probably be on out of memory or whatever so I guess it's the best thing we can do?

Reply:
I would do an expect then, just to point out to the correct place to check in case of a potential future bug report

Reply:
I mean, unwraps always print the filename and line, right? Is there a reason this needs to be an expect and the other 56 unwraps not?

```
oo7 $ git g 'unwrap()' | wc -l
57
```